src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseDataMapper.java

```diff
@@ -0,0 +1,169 @@
+package uk.gov.hmcts.reform.prl.controllers.citizen;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.prl.enums.ChildArrangementOrderTypeEnum;
+import uk.gov.hmcts.reform.prl.enums.OrderTypeEnum;
+import uk.gov.hmcts.reform.prl.enums.YesOrNo;
+import uk.gov.hmcts.reform.prl.enums.citizen.CourtOrderTypeEnum;
+import uk.gov.hmcts.reform.prl.models.c100rebuild.C100RebuildCourtOrderElements;
+import uk.gov.hmcts.reform.prl.models.c100rebuild.C100RebuildHearingWithoutNoticeElements;
+import uk.gov.hmcts.reform.prl.models.c100rebuild.C100RebuildInternationalElements;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.prl.enums.ChildArrangementOrderTypeEnum.bothLiveWithAndSpendTimeWithOrder;
+import static uk.gov.hmcts.reform.prl.enums.ChildArrangementOrderTypeEnum.liveWithOrder;
+import static uk.gov.hmcts.reform.prl.enums.ChildArrangementOrderTypeEnum.spendTimeWithOrder;
+import static uk.gov.hmcts.reform.prl.enums.OrderTypeEnum.childArrangementsOrder;
+import static uk.gov.hmcts.reform.prl.enums.OrderTypeEnum.prohibitedStepsOrder;
+import static uk.gov.hmcts.reform.prl.enums.OrderTypeEnum.specificIssueOrder;
+import static uk.gov.hmcts.reform.prl.enums.YesOrNo.No;
+import static uk.gov.hmcts.reform.prl.enums.YesOrNo.Yes;
+
+@Component
+public class CaseDataMapper {
```

Comment:
i feel we should refactor this class and move the mappers for each feature in a separate class

Reply:
Done